### ISSUE TITLE
Invoke-WebRequest Headers not treated as Dictionary[string,string]

### ISSUE THREAD

**Issue by daniel0x00** [PowerShell/PowerShell#6674]
`Invoke-WebRequest` Headers property is handled as a `Dictionary[string,System.Collections.Generic.IEnumerable[string]]` in PS Core; while in Windows PS is `Dictionary[string,string]`. 

![2018-04-18 13_01_54-c__program files_powershell_6 1 0-preview 1_pwsh exe](https://user-images.githubusercontent.com/6474092/38928350-d50decfa-4308-11e8-8e06-5b65400281fe.png)

Therefore the Values of the headers are treated differently:

![daqudfiwkaeg6fp](https://user-images.githubusercontent.com/6474092/38928587-951ef07a-4309-11e8-89cf-560de4f41ba2.jpg)

Steve Lee pointed out this could not be expected:
[https://twitter.com/daniel0x00/status/984894285387632641](url)

Thanks,
Daniel Ferreira

**Comment by SteveL-MSFT** [comment]
@daniel0x00 I looked into the history of this.  It appears that at the time the webcmdlets were ported to dotnetcore 1.0 for NanoServer, the type changed for Headers.  At this time, it would be a breaking change to make it consistent with Windows PowerShell.  Given the extensive changes in the webcmdlets, I think you should consider it a different cmdlet with the same name as the Windows PowerShell counterpart.  Compatibility is retained where possible, but in this case, it seems we need to worry about compatibility with PS Core 6.0.

**Comment by markekraus** [comment]
For context, This is `By Design` and is documented in the [breaking changes](https://github.com/PowerShell/PowerShell/blob/master/docs/BREAKINGCHANGES.md#changes-to-web-cmdlets).

I consider reverting it to the same behavior as Windows PowerShell, but this breaking change actually fixes unexpected behavior. In Windows PowerShell, when a remote endpoint returns multiple response headers with the same name, they were being concatenated and joined with a comma. HttpClient treats these as separate array entries. The old method made it really hard to parse complex headers which include header separator characters as part of their values (such as cookies).

If the endpoint returns 

```
MyHeader: A,B,C
MyHeader: D,E,F
```

In Windows PowerShell `$result.Headers.MyHeader` would be `A,B,C,D,E,F`. which makes it impossible to determine the separate values. In PowerShell Core `$result.Headers.MyHeader[0]` would be `A,B,C` and `$result.Headers.MyHeader[1]` would be `D,E,F`.

You can see this difference in behavior with this:

```powershell
$Result = Invoke-WebRequest 'https://httpbin.org/response-headers?MyHeader=A,B,C&MyHeader=D,E,F'
$Result.Headers.MyHeader | Select -First 1
$Result.Headers.MyHeader | Select -Skip 1
```

[More info.](https://get-powershellblog.blogspot.com/2017/11/powershell-core-web-cmdlets-in-depth.html#L10)

**Comment by daniel0x00** [comment]
@markekraus now that I got it, I will vote for leave it as is in PS Core, so no major changes will happen again. New string[] way makes more sense even if that means code needs to be rewritten (as mine). Thanks for the effort of explaining!
